.storybook/symbol-doc-loader.js

```diff
@@ -0,0 +1,52 @@
+// @ts-check
+const {createDocProgram} = require('../modules/docs/docgen/createDocProgram');
+const routes = require('./routes');
+
+const routeKeys = Object.keys(routes);
+
+// Tracks files that have been processed. If a file is already processed, it
+// means the file has been updated and we need to update the Typescript program
+// so the changes are reflected in our doc output. If we don't update the TS
+// program, the docs will be processed with the outdated cache of the file
+// contents.
+const filesProcessedMap = new Map();
+
+/** @typedef {import('webpack').loader.Loader} Loader */
+/** @typedef {import('webpack').loader.LoaderContext} LoaderContext */
+
+let {parser, update} = createDocProgram();
+
+/**
+ * @this {LoaderContext}
+ * @param {Parameters<Loader>[0]} source
+ */
+function symbolDocLoader(source) {
+  if (filesProcessedMap.has(this.resourcePath)) {
+    parser = update();
```

Comment:
`update` updates the Typescript program. A Typescript program doesn't hot reload. I looked into a faster, more efficient version of updates using the Typescript Language Service that can handle per-AST not updates, but that was more complicated that I was willing to deal with. This means updating JSDoc or types may take about 1.5s to reflect in Storybook vs a few ms. We may fix this later.